dali_tf_plugin/dali_tf_plugin_install_tool.py

```diff
@@ -243,6 +223,86 @@ def install(self):
                 raise ImportError("Error while importing the DALI TF plugin built from source, will not install")
             print("Installation successful")
 
+    def check_install_env(self):
+        print("Checking build environment for DALI TF plugin ...")
+        print(self.debug_str())
+
+        Path(self.plugin_dest_dir).mkdir(parents=True, exist_ok=True)
+
+        if not self.tf_version or not self.tf_path:
+            error_msg = "Installation error:"
+            error_msg += "\n Tensorflow installation not found. Install `tensorflow-gpu` and try again"
+            error_msg += '\n' + self.debug_str()
+            raise ImportError(error_msg)
+
+    def test(self):
+        from nvidia.dali.pipeline import pipeline_def
+        import nvidia.dali.types as types
+        import nvidia.dali.plugin.tf as dali_tf
+        import tensorflow as tf
+        try:
+            from tensorflow.compat.v1 import Session
+        except Exception:
+            # Older TF versions don't have compat.v1 layer
+            from tensorflow import Session
+
+        try:
+            tf.compat.v1.disable_eager_execution()
+        except Exception:
+            pass
+
+        @pipeline_def()
+        def get_dali_pipe():
+            data = types.Constant(1)
+            return data
+
+        def get_data():
+            batch_size = 3
+            pipe = get_dali_pipe(batch_size=batch_size, device_id=types.CPU_ONLY_DEVICE_ID, num_threads=1)
+            daliop = dali_tf.DALIIterator()
+            out = []
+            with tf.device('/cpu'):
+                data = daliop(pipeline = pipe,
+                              shapes = [(batch_size)],
+                              dtypes = [tf.int32],
+                              device_id = types.CPU_ONLY_DEVICE_ID)
+                out.append(data)
+            return [out]
+
+        test_batch = get_data()
+        with Session() as sess:
+            print(sess.run(test_batch))
+
+    def test_if_dali_available(self):
+        try:
+            self.test()
+        except ImportError as e:
+            print("DALI is not available yet. Skipping tests...")
+
+    def install(self):
+        self.check_install_env()
+
+        if self.prebuilt_exact_ver:
+            self.install_prebuilt()
+            try:
+                self.test_if_dali_available()
+                return  # We can finish the installation here
+            except Exception as e:
+                print("Error while testing the prebuilt plugin: ", e)
+                # Will try building from source
+
+        try:
+            self.build()
+        except Exception as e:
+            print("Build from source failed with error: ", e)
+            if self.can_install_prebuilt:
+                print("Trying to install prebuilt plugin")
+                self.install_prebuilt()
```

Comment:
So if `prebuilt_exact_ver` and `can_install_prebuilt` are True, we may have three attempts at installing DALI: `install_prebuilt`, `build` and again `install_prebuilt`?

Reply:
True. Perhaps we should only try the second install prebuilt if there was no first attempt.

Reply:
Done